test/models/badge_test.rb

```diff
@@ -2,8 +2,10 @@
 
 class BadgeTest < ActiveSupport::TestCase
   test "percentage_awardees" do
-    badge = create :badge, num_awardees: 123_456
-    assert_equal 15.44, badge.percentage_awardees
+    create_list(:user, 21)
+
+    badge = create :badge, num_awardees: 4
```

Comment:
For some reason I'm now getting this error:

```shell
ActiveRecord::RecordNotUnique: Mysql2::Error: Duplicate entry 'Rookie' for key 'badges.index_badges_on_name'
    test/models/badge_test.rb:7:in `block in <class:BadgeTest>'

rails test test/models/badge_test.rb:4
```

I'm not exactly sure how my changes are causing this to happen. Thoughts?

Reply:
I suspect you need to create the badge before creating the users.